crypto/x509/by_dir.c

```diff
@@ -257,165 +258,180 @@ static int get_cert_by_subject_ex(X509_LOOKUP *xl, X509_LOOKUP_TYPE type,
     }
 
     ctx = (BY_DIR *)xl->method_data;
-    h = X509_NAME_hash_ex(name, libctx, propq, &i);
-    if (i == 0)
-        goto finish;
-    for (i = 0; i < sk_BY_DIR_ENTRY_num(ctx->dirs); i++) {
-        BY_DIR_ENTRY *ent;
-        int idx;
-        BY_DIR_HASH htmp, *hent;
-
-        ent = sk_BY_DIR_ENTRY_value(ctx->dirs, i);
-        j = strlen(ent->dir) + 1 + 8 + 6 + 1 + 1;
-        if (!BUF_MEM_grow(b, j)) {
-            ERR_raise(ERR_LIB_X509, ERR_R_BUF_LIB);
-            goto finish;
+
+    hash[0] = X509_NAME_hash_ex(name, libctx, propq, &l); /* SHA1 name hash */
+    if (l == 0) {
+        hash[0] = 0;
+    }
+#ifndef OPENSSL_NO_MD5    
+    hash[1] = X509_NAME_hash_old(name); /* MD5 name hash */
+#else
+    hash[1] = 0;
+#endif
+
+    for (l = 0; l < 2; ++l) {
+        h = hash[l];
+        if (h == 0) {
+            continue; /* skip invalid hash */
         }
```

Comment:
Please do not add `{}` around a single statement.

Reply:
4612aaf293430102137904d73096f496e642fe43